### PR TITLE
chore: package example as a snap

### LINKED ISSUE #135: Package example as snap
I think it would be nice to have the example app of yaru_icons packaged as a snap, like yaru_widgets.

### PR DESCRIPTION
**Pull request by Jupi007**
@kenvandine I simply copied that you did here: https://github.com/ubuntu/yaru_widgets.dart/pull/367
But I'd like to have your review and your help to publish it.

Fixes #135

### CODE REVIEW

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+apps:
+  yaru-icons-example:
+    command: bin/yaru_icons_example
+    desktop: snap/gui/yaru-icons-example.desktop
+    extensions: [gnome]
+    plugs:
+      - gsettings
+      - network
+      - network-manager-observe
```
Comment (jpnurmi) [inline snap/snapcraft.yaml:58]: does the example need network?

Comment (Jupi007) [reply snap/snapcraft.yaml:58]: Not at all, indeed 😅

Comment (jpnurmi) [reply snap/snapcraft.yaml:58]: ok, let's leave these two lines out. otherwise looks good to me

Comment (Jupi007) [reply snap/snapcraft.yaml]: Done, thanks for the review :)

**Comment by jpnurmi** [comment]
There's a mismatch (`example` vs. `yaru_icons_example`) in the name of the binary:

```
Cannot pack snap file: Command '['snap', 'pack', '--filename', 'yaru-icons-example_0+git.327401c_amd64.snap', '--compression', 'xz', PosixPath('/root/prime'), PosixPath('/root/project')]' returned non-zero exit status 1. (2023/06/30 07:08:25.354155 container.go:215: in snap "yaru-icons-example": path "bin/yaru_icons_example" does not exist
error: cannot pack "/root/prime": snap is unusable due to missing files)                                                                                               
Failed to execute pack in instance.                                                                                                                                      
```

Either tell snapcraft that the binary is `bin/example`:

```diff
diff --git a/snap/snapcraft.yaml b/snap/snapcraft.yaml
index a9d9875..b9d38c3 100644
--- a/snap/snapcraft.yaml
+++ b/snap/snapcraft.yaml
@@ -49,7 +49,7 @@ parts:
 
 apps:
   yaru-icons-example:
-    command: bin/yaru_icons_example
+    command: bin/example
     desktop: snap/gui/yaru-icons-example.desktop
     extensions: [gnome]
     plugs:
```

Or rename the binary to `yaru_icons_example`:

```diff
diff --git a/example/linux/CMakeLists.txt b/example/linux/CMakeLists.txt
index a558bc4..9ffa4f2 100644
--- a/example/linux/CMakeLists.txt
+++ b/example/linux/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.10)
 project(runner LANGUAGES CXX)
 
-set(BINARY_NAME "example")
+set(BINARY_NAME "yaru_icons_example")
 set(APPLICATION_ID "com.example.example")
 
 cmake_policy(SET CMP0063 NEW)
```

**Comment by Jupi007** [comment]
@jpnurmi I just checked the `yaru_widgets` repo and the binary has been renamed. So I replicated that here.

Btw, I also changed the `APPLICATION_ID`, is it fine?

**Review by jpnurmi — APPROVED** [review]
LGTM :+1: 

```sh
$ snapcraft 
Launching instance...                                                                                                                                                    
...                                                                                                                             
Version has been set to '0+git.e38151b'                                                                                                                                  
Created snap package yaru-icons-example_0+git.e38151b_amd64.snap
$ snap install yaru-icons-example_0+git.e38151b_amd64.snap --dangerous
yaru-icons-example 0+git.e38151b installed
$ yaru-icons-example
```
Works :)

**Comment by Jupi007** [comment]
Awesome, thanks.
Let's merge this.

I had another question: I looked at the snapcraft doc, and I didn't saw how to do automated uploads. Is it possible?

**Comment by jpnurmi** [comment]
> I had another question: I looked at the snapcraft doc, and I didn't saw how to do automated uploads. Is it possible?

On snapcraft.io, you can link a snap to a GitHub repo. :)

**Comment by Jupi007** [comment]
Oh okay, great!
Thanks :)

**Pull request merged**